client/components/activity-card/index.jsx

```diff
@@ -170,6 +170,13 @@ class ActivityCard extends Component {
 			? this.state.showTopPopoverMenu
 			: this.state.showBottomPopoverMenu;
 
+		// The activity itself may not be rewindable, but at least one of the
+		// streams should be; if this is the case, make sure we send the user
+		// to a valid restore/download point when they click an action button
+		const actionableRewindId = activity.activityIsRewindable
+			? activity.rewindId
+			: activity.streams.find( ( s ) => s.activityIsRewindable )?.rewindId;
```

Comment:
Would it make sense to memoize this? How many streams could a single activity have?

Reply:
I believe it's limited to 10 streams by default in the API, so not a huge amount.